modules/core/src/main/java/com/google/refine/expr/Evaluable.java

```diff
@@ -47,4 +49,19 @@ public interface Evaluable {
      * @return
      */
     public Object evaluate(Properties bindings);
+
+    /**
+     * Returns the names of the columns this expression depends on.
```

Comment:
```suggestion
     * Returns a best guess of the names of the columns this expression depends on. NOTE: This has the following limitations: ...
```

I think it would be worth expanding this with the known limitations, so that potential callers have their expectations set correctly.